opm/simulators/linalg/bda/rocm/rocsparseSolverBackend.cpp

```diff
@@ -151,26 +151,55 @@ gpu_pbicgstab([[maybe_unused]] WellContributions<Scalar>& wellContribs,
 
 // HIP_VERSION is defined as (HIP_VERSION_MAJOR * 10000000 + HIP_VERSION_MINOR * 100000 + HIP_VERSION_PATCH)
 #if HIP_VERSION >= 60000000
-    ROCSPARSE_CHECK(rocsparse_dbsrmv(handle, dir, operation,
-                                     Nb, Nb, nnzb, &one, descr_A,
-                                     d_Avals, d_Arows, d_Acols, block_size,
-                                     spmv_info, d_x, &zero, d_r));
+    if constexpr (std::is_same_v<Scalar,float>) {
```

Comment:
Include `<type_traits>` in this file because `is_same_v` is used

Reply:
done.